test/protocol/produce_test.exs

```diff
@@ -10,7 +10,7 @@ defmodule KafkaEx.Protocol.Produce.Test do
       ]
     })
 
-    assert expected_request == request
+    assert expected_request == :erlang.iolist_to_binary(request)
```

Comment:
I would prefer that we just change `expected_result` in these tests to a list, since that is the expected result.